File: HW/HW10/Main.java
public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ans = new ArrayList<List<Integer>>();
        Queue<TreeNode> q = new LinkedList<>();

        if (root == null) {
            return ans;
        }

        q.add(root);
        while (!q.isEmpty()) {
            int length = q.size();
            List<Integer> level = new ArrayList<>();
            for (int i = 0; i < length; i++) {
                if (q.peek().left != null) {
                    q.add(q.peek().left);
                }
                if (q.peek().right != null) {
                    q.add(q.peek().right);
                }
                level.add(q.remove().val);
            }
            ans.add(level);
        }
        return ans;
    }

File: HW/HW10/Main.py
def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
    ans = []

    q = collections.deque()
    q.append(root)

    while q:
        length = len(q)
        level = []
        for i in range(length):
            n = q.popleft()
            if n:
                level.append(n.val)
                q.append(n.left)
                q.append(n.right)
        if level:
            ans.append(level)
    return ans
